shell/platform/android/io/flutter/view/FlutterMain.java

```diff
@@ -308,4 +308,30 @@ public static String findAppBundlePath(Context applicationContext) {
         File appBundle = new File(dataDirectory, sFlutterAssetsDir);
         return appBundle.exists() ? appBundle.getPath() : null;
     }
+
+    /**
+     * Returns the file name for the given asset.
+     * The returned file name can be used to access the asset in the APK
+     * through the {@link AssetManager} API.
+     *
+     * @param asset the name of the asset. The name can be hierarchical
+     * @return      the file name to be used with {@link AssetManager}
+     */
+    public static String getLookupKeyForAsset(String asset) {
+        return sFlutterAssetsDir + File.separator + asset;
+    }
+
+    /**
+     * Returns the file name for the given asset which originates from the
+     * specified packageName. The returned file name can be used to access
+     * the asset in the APK through the {@link AssetManager} API.
+     *
+     * @param asset       the name of the asset. The name can be hierarchical
+     * @param packageName the name of the package from which the asset originates
+     * @return            the file name to be used with {@link AssetManager}
+     */
+    public static String getLookupKeyForAsset(String asset, String packageName) {
```

Comment:
How "official" is the convention with /packages/? It seems AssetImage is doing this calculation itself, and AssetBundle doesn't know anything about it. It seems weird that this hook knows about stuff that AssetBundle doesn't.
Should we have this logic in AssetBundle as well, or in general leave it up to clients to figure out?